### ISSUE TITLE
feat(query): use `__call` for query items

### ISSUE THREAD

**Issue by cbochs** [cbochs/portal.nvim#5]
Right now query items are registered with a `Portal.Predicate` under a `predicate` table value. This means that to invoke any query item, one must call `{query_item}.predicate(jump)`. This can be cumbersome and probably not proper hiding of the implementation details.

Rather, a query item should have its metatable set with a simple `__call` function that defers to the query item's `predicate`. This should keep the query item relatively simple to create, but allow for calling a query item's predicate directly.

In short, the following would happen:

```
require("portal.query").valid.predicate(jump)
```

would be shortened to 

```lua
require("portal.query").valid(jump)
```

This makes calling a query item within another query item less complex, but keeps the simple interface of a `Portal.Query`.

**Comment by cbochs** [comment]
Resolved by 0c4f19a65b171eb44033718058ffe2128be835c9
